src/main/io/gps.c

```diff
@@ -683,13 +683,18 @@ void gpsUpdate(timeUs_t currentTimeUs)
     timeUs_t executeTimeUs;
     gpsState_e gpsCurState = gpsData.state;
 
-
     // read out available GPS bytes
     if (gpsPort) {
-        while (serialRxBytesWaiting(gpsPort) && (cmpTimeUs(micros(), currentTimeUs) < GPS_MAX_WAIT_DATA_RX)) {
+        while (serialRxBytesWaiting(gpsPort)) {
+            if (cmpTimeUs(micros(), currentTimeUs) > GPS_MAX_WAIT_DATA_RX) {
+                // Wait 1ms and come back
+                rescheduleTask(TASK_SELF, 1000);
+                return;
+            }
             gpsNewData(serialRead(gpsPort));
         }
-    } else if (GPS_update & GPS_MSP_UPDATE) { // GPS data received via MSP
+        rescheduleTask(TASK_SELF, TASK_PERIOD_HZ(100));
```

Comment:
Maybe we should be using a define for the task frequency when we're setting it in multiple places? If for some reason we wanted to change it in the future we would need to do some debugging to figure out why it just changes itself back to 100Hz

Reply:
Done, see `TASK_GPS_RATE` and `TASK_GPS_RATE_FAST` in `gps.h`.